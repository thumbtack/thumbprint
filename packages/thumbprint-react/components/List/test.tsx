import React from 'react';
import { mount } from 'enzyme';
import { List, ListItem } from './index';

test('renders content that is passed in', () => {
    const wrapper = mount(<List>goose</List>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

test('renders ul for bullet theme', () => {
    const wrapper = mount(<List theme="bullet">goose</List>);
    expect(wrapper.find('ul')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});

test('renders ol for decimal theme', () => {
    const wrapper = mount(<List theme="decimal">goose</List>);
    expect(wrapper.find('ol')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});

test('renders ol for alpha theme', () => {
    const wrapper = mount(<List theme="alpha">goose</List>);
    expect(wrapper.find('ol')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});

test('item renders content that is passed in', () => {
    const wrapper = mount(<ListItem>goose</ListItem>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});
