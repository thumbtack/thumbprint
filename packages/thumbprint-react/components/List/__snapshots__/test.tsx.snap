// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item renders content that is passed in 1`] = `
<ListItem>
  <li
    className="item"
  >
    goose
  </li>
</ListItem>
`;

exports[`renders content that is passed in 1`] = `
<List>
  <ul
    className="root bullet"
  >
    goose
  </ul>
</List>
`;

exports[`renders ol for alpha theme 1`] = `
<List
  theme="alpha"
>
  <ol
    className="root alpha"
  >
    goose
  </ol>
</List>
`;

exports[`renders ol for decimal theme 1`] = `
<List
  theme="decimal"
>
  <ol
    className="root decimal"
  >
    goose
  </ol>
</List>
`;

exports[`renders ul for bullet theme 1`] = `
<List
  theme="bullet"
>
  <ul
    className="root bullet"
  >
    goose
  </ul>
</List>
`;
